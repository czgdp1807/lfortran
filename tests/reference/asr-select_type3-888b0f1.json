{
    "basename": "asr-select_type3-888b0f1",
    "cmd": "lfortran --indent --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/select_type3.f90",
    "infile_hash": "448ba5f04a53ba15b8b73bf5b30478062b3f3682965327d5b2bf9579",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-select_type3-888b0f1.stdout",
    "stdout_hash": "a8c63f128fd2f32736fa0ec7aca97bdac91744550beecf183b9118f6",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}