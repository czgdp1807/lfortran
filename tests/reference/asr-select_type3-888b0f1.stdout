(TranslationUnit
    (SymbolTable
        1
        {
            class_default_select_type:
                (Module
                    (SymbolTable
                        2
                        {
                            check_table:
                                (Function
                                    (SymbolTable
                                        7
                                        {
                                            tab:
                                                (Variable
                                                    7
                                                    tab
                                                    []
                                                    Out
                                                    ()
                                                    ()
                                                    Default
                                                    (Class
                                                        2 toml_table
                                                        []
                                                    )
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    check_table
                                    (FunctionType
                                        [(Class
                                            2 toml_table
                                            []
                                        )]
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 7 tab)]
                                    []
                                    ()
                                    Source
                                    Public
                                    Implementation
                                    ()
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    []
                                    []
                                    .false.
                                ),
                            enum_stat:
                                (StructType
                                    (SymbolTable
                                        5
                                        {
                                            fatal:
                                                (Variable
                                                    5
                                                    fatal
                                                    []
                                                    Local
                                                    (IntegerUnaryMinus
                                                        (IntegerConstant 1 (Integer 4 []))
                                                        (Integer 4 [])
                                                        (IntegerConstant -1 (Integer 4 []))
                                                    )
                                                    ()
                                                    Save
                                                    (Integer 4 [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            success:
                                                (Variable
                                                    5
                                                    success
                                                    []
                                                    Local
                                                    (IntegerConstant 0 (Integer 4 []))
                                                    ()
                                                    Save
                                                    (Integer 4 [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    enum_stat
                                    []
                                    [success
                                    fatal]
                                    Source
                                    Public
                                    .false.
                                    ()
                                    ()
                                ),
                            get_table:
                                (Function
                                    (SymbolTable
                                        6
                                        {
                                            1_enum_stat_fatal:
                                                (ExternalSymbol
                                                    6
                                                    1_enum_stat_fatal
                                                    5 fatal
                                                    enum_stat
                                                    []
                                                    fatal
                                                    Public
                                                ),
                                            present:
                                                (ExternalSymbol
                                                    6
                                                    present
                                                    10 present
                                                    lfortran_intrinsic_builtin
                                                    []
                                                    present
                                                    Private
                                                ),
                                            ptr:
                                                (Variable
                                                    6
                                                    ptr
                                                    []
                                                    Out
                                                    ()
                                                    ()
                                                    Default
                                                    (Pointer
                                                        (Struct
                                                            2 toml_table
                                                            []
                                                        )
                                                    )
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            stat:
                                                (Variable
                                                    6
                                                    stat
                                                    []
                                                    Out
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4 [])
                                                    Source
                                                    Public
                                                    Optional
                                                    .false.
                                                ),
                                            table:
                                                (Variable
                                                    6
                                                    table
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (Class
                                                        2 toml_table
                                                        []
                                                    )
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            tmp:
                                                (Variable
                                                    6
                                                    tmp
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Pointer
                                                        (Class
                                                            2 toml_value
                                                            []
                                                        )
                                                    )
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            ~select_type_block_:
                                                (Block
                                                    (SymbolTable
                                                        8
                                                        {
                                                            1_enum_stat_success:
                                                                (ExternalSymbol
                                                                    8
                                                                    1_enum_stat_success
                                                                    5 success
                                                                    enum_stat
                                                                    []
                                                                    success
                                                                    Public
                                                                ),
                                                            present:
                                                                (ExternalSymbol
                                                                    8
                                                                    present
                                                                    10 present
                                                                    lfortran_intrinsic_builtin
                                                                    []
                                                                    present
                                                                    Private
                                                                )
                                                        })
                                                    ~select_type_block_
                                                    [(=>
                                                        (Var 6 ptr)
                                                        (Var 6 tmp)
                                                    )
                                                    (If
                                                        (FunctionCall
                                                            8 present
                                                            ()
                                                            [((Var 6 stat))]
                                                            (Logical 4 [])
                                                            ()
                                                            ()
                                                        )
                                                        [(=
                                                            (Var 6 stat)
                                                            (StructInstanceMember
                                                                (Var 2 toml_stat)
                                                                8 1_enum_stat_success
                                                                (Integer 4 [])
                                                                (IntegerConstant 0 (Integer 4 []))
                                                            )
                                                            ()
                                                        )]
                                                        []
                                                    )]
                                                )
                                        })
                                    get_table
                                    (FunctionType
                                        [(Class
                                            2 toml_table
                                            []
                                        )
                                        (Pointer
                                            (Struct
                                                2 toml_table
                                                []
                                            )
                                        )
                                        (Integer 4 [])]
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    [check_table
                                    present]
                                    [(Var 6 table)
                                    (Var 6 ptr)
                                    (Var 6 stat)]
                                    [(Nullify
                                        [6 ptr]
                                    )
                                    (If
                                        (PointerAssociated
                                            (Var 6 tmp)
                                            ()
                                            (Logical 4 [])
                                            ()
                                        )
                                        [(SelectType
                                            [(TypeStmtName
                                                2 toml_table
                                                [(BlockCall
                                                    -1
                                                    6 ~select_type_block_
                                                )]
                                            )]
                                            [(If
                                                (FunctionCall
                                                    6 present
                                                    ()
                                                    [((Var 6 stat))]
                                                    (Logical 4 [])
                                                    ()
                                                    ()
                                                )
                                                [(=
                                                    (Var 6 stat)
                                                    (StructInstanceMember
                                                        (Var 2 toml_stat)
                                                        6 1_enum_stat_fatal
                                                        (Integer 4 [])
                                                        (IntegerConstant -1 (Integer 4 []))
                                                    )
                                                    ()
                                                )]
                                                []
                                            )]
                                        )]
                                        [(SubroutineCall
                                            2 check_table
                                            ()
                                            [((Var 6 table))]
                                            ()
                                        )]
                                    )]
                                    ()
                                    Source
                                    Public
                                    Implementation
                                    ()
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    []
                                    []
                                    .false.
                                ),
                            toml_stat:
                                (Variable
                                    2
                                    toml_stat
                                    []
                                    Local
                                    ()
                                    ()
                                    Parameter
                                    (Struct
                                        2 enum_stat
                                        []
                                    )
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            toml_table:
                                (StructType
                                    (SymbolTable
                                        4
                                        {
                                            implicit:
                                                (Variable
                                                    4
                                                    implicit
                                                    []
                                                    Local
                                                    (LogicalConstant
                                                        .false.
                                                        (Logical 4 [])
                                                    )
                                                    ()
                                                    Save
                                                    (Logical 4 [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            inline:
                                                (Variable
                                                    4
                                                    inline
                                                    []
                                                    Local
                                                    (LogicalConstant
                                                        .false.
                                                        (Logical 4 [])
                                                    )
                                                    ()
                                                    Save
                                                    (Logical 4 [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    toml_table
                                    []
                                    [implicit
                                    inline]
                                    Source
                                    Public
                                    .false.
                                    ()
                                    2 toml_value
                                ),
                            toml_value:
                                (StructType
                                    (SymbolTable
                                        3
                                        {
                                            float:
                                                (Variable
                                                    3
                                                    float
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 4 [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            int:
                                                (Variable
                                                    3
                                                    int
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4 [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    toml_value
                                    []
                                    [int
                                    float]
                                    Source
                                    Public
                                    .false.
                                    ()
                                    ()
                                )
                        })
                    class_default_select_type
                    [lfortran_intrinsic_builtin]
                    .false.
                    .false.
                ),
            lfortran_intrinsic_builtin:
                (IntrinsicModule lfortran_intrinsic_builtin)
        })
    []
)
